### PR TITLE
Initial addition of CODEOWNERS file

### PR DESCRIPTION
**Pull request by leahwicz**
### Description

This file will help us automatically map files to the correct sub-teams for review. This file will be every growing and evolving so this is by no means a finished version. Just looking to get it started and then add to it as we go.

### Checklist

- [ ] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [ ] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have updated the `CHANGELOG.md` and added information about my change

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# Changes to GitHub configurations including Actions
+/.github/ @leahwicz
+
+# Adapter / plugin code
+/plugins @dbt-labs/core-adapters
+
+# Perf regression testing framework
+# This excludes the test project files itself since those aren't specific 
+# framework changes (excluded by not setting an owner next to it- no owner)
+/performance @nathaniel-may
```
Comment (leahwicz) [inline .github/CODEOWNERS:21]: @nathaniel-may I put you down as the reviewer here for now until we ramp other people up here (which would be the goal). I hope that works for you but if not, let me know

**Comment by nathaniel-may** [comment]
you can probably add the event module since it's pretty cleanly segmented:
/core/dbt/events - language team (def emily gerda me at least)

**Comment by leahwicz** [comment]
@nathaniel-may I added both teams since in the log run, logging will fall under execution but for now the expertise lies with language

**Review by emmyoop — APPROVED** [review]
LGTM

**Review by nathaniel-may — APPROVED** [review]
looks like a good start to me!

**Review by VersusFacit — APPROVED** [review]
👍 Although the token-matcher in me is bothered that the `*` comes first since all matches will be against that and never fall through to your specific cases. 😂

**Comment by nathaniel-may** [comment]
@VersusFacit good point. I have no idea what the semantics of line order are in this file 🤔

**Comment by leahwicz** [comment]
@nathaniel-may @VersusFacit - you're right, it's weird but the last statement takes precedence over everything above it so that's why the `*` has to be at the top (according to the GitHub docs and example)

https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#example-of-a-codeowners-file

**Comment by leahwicz** [comment]
Maybe I'll add that as a comment

**Pull request merged**